app/services/proceeding_type_service.rb

```diff
@@ -0,0 +1,96 @@
+class ProceedingTypeService
+  class ProceedingTypeServiceError < StandardError; end
+
+  def self.call(ccms_code)
+    new(ccms_code).call
+  end
+
+  def initialize(ccms_code)
+    @ccms_code = ccms_code
+    @response = create_skeleton_response
+  end
+
+  def call
+    raise ProceedingTypeServiceError, 'Must specify a proceeding type' if @ccms_code.nil?
+
+    populate_response(@ccms_code)
+    @response
+  rescue StandardError => err
+    @response = error_response_for(err)
+  end
+
+  private
+
+  def populate_response(ccms_code)
+    pt = ProceedingType.find_by!(ccms_code: ccms_code)
+    add_proceeding_type_to_response(pt)
+    add_cost_limitations_to_response(pt)
+    add_scope_limitations_to_response(pt)
+  end
+
+  def add_proceeding_type_to_response(proceeding_type) # rubocop:disable Metrics/AbcSize
+    @response[:ccms_code] = proceeding_type.ccms_code
+    @response[:meaning] = proceeding_type.meaning
+    @response[:ccms_category_law_code] = proceeding_type.ccms_category_law_code
+    @response[:ccms_matter_code] = proceeding_type.ccms_matter_code
+    @response[:description] = proceeding_type.description
+    @response[:ccms_category_law] = proceeding_type.matter_type.category_of_law
+    @response[:ccms_matter] = proceeding_type.matter_type.name
+  end
```

Comment:
It looks like you can make rubocop happy by doing this:

```suggestion
  def add_proceeding_type_to_response(proceeding_type)
    matter_type = proceeding_type.matter_type
    @response[:ccms_code] = proceeding_type.ccms_code
    @response[:meaning] = proceeding_type.meaning
    @response[:ccms_category_law_code] = proceeding_type.ccms_category_law_code
    @response[:ccms_matter_code] = proceeding_type.ccms_matter_code
    @response[:description] = proceeding_type.description
    @response[:ccms_category_law] = matter_type.category_of_law
    @response[:ccms_matter] = matter_type.name
  end
```

Reply:
thanks, that does indeed work